### DISCUSSION TITLE
Random Decorator

### DISCUSSION THREAD

**Discussion by RedstoneParadox** [bitbrain/beehave#273] [Ideas]
This would be a decorator that would have a random chance to return `FAILURE` instead of executing its child node(s). This is something I'd be willing to contribute myself.

Edit: I suppose you could probably already accomplish this with the random selector composite by just having one child that always returns `FAILURE`, but I believe this would be a cleaner approach.

**Comment by bitbrain** [comment]
> Edit: I suppose you could probably already accomplish this with the random selector composite by just having one child that always returns FAILURE, but I believe this would be a cleaner approach.

given there is a solution for this already, I do not see the need for a dedicated node. It sounds like a very niche edge case and the nodes provided by Beehave ideally should be general purpose, not specific.
